akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala

```diff
@@ -177,6 +179,9 @@ private[http] class HttpRequestParser(_settings: ParserSettings,
             parseFixedLengthBody(contentLength, closeAfterResponseCompletion)(input, bodyStart)
           }
 
+        case Some(_) if !method.isEntityAccepted ⇒
```

Comment:
Yes, make take would be so as well. The `Transfer-Encoding` header always contains meta info on the message entity. So, if there is no entity a `Transfer-Encoding` header makes no sense.